packages/wrangler/src/__tests__/configuration.test.ts

```diff
@@ -728,12 +728,85 @@ describe("normalizeAndValidateConfig()", () => {
           - Expected \\"jsx_factory\\" to be of type string but got 999.
           - Expected \\"jsx_fragment\\" to be of type string but got 1000.
           - Expected \\"tsconfig\\" to be of type string but got true.
-          - Expected \\"name\\" to be of type string but got 111.
+          - Expected \\"name\\" to be of type string, alphanumeric and lowercase with dashes only but got 111.
           - Expected \\"main\\" to be of type string but got 1333.
           - Expected \\"usage_model\\" field to be one of [\\"bundled\\",\\"unbound\\"] but got \\"INVALID\\"."
       `);
     });
 
+    it("should error on invalid `name` value with spaces", () => {
+      const expectedConfig: RawEnvironment = {
+        name: "NCC 1701 D",
+      } as unknown as RawEnvironment;
+
+      const { config, diagnostics } = normalizeAndValidateConfig(
+        expectedConfig,
+        undefined,
+        { env: undefined }
+      );
+
+      expect(config).toEqual(expect.objectContaining(expectedConfig));
+      expect(diagnostics.hasWarnings()).toBe(false);
+      expect(diagnostics.renderErrors()).toMatchInlineSnapshot(`
+        "Processing wrangler configuration:
+          - Expected \\"name\\" to be of type string, alphanumeric and lowercase with dashes only but got \\"NCC 1701 D\\"."
+      `);
+    });
+
+    it("should be valid `name` with underscores", () => {
+      const expectedConfig: RawEnvironment = {
+        name: "ecnterprise_nx_01",
```

Comment:
```suggestion
        name: "enterprise_nx_01",
```

Or is the `c` there on purpose?

Reply:
lol